src/server/routes/editor.js

```diff
@@ -325,18 +348,18 @@ router.get('/:id/achievements', (req, res, next) => {
 
 	Promise.join(achievementJSONPromise, editorJSONPromise,
 		(achievementJSON, editorJSON) => {
-			const markup =
-				ReactDOMServer.renderToString(AchievementForm({
-					editor: editorJSON,
-					achievement: achievementJSON
-				}));
-
-			res.render('editor/achievements', {
-				props: {
-					editor: editorJSON,
-					achievement: achievementJSON
-				},
-				markup
+			const props = Object.assign({}, req.app.locals, res.locals, {
+				editor: editorJSON,
+				achievement: achievementJSON
+			});
+			res.render('target', {
```

Comment:
Hmm, generally I think I prefer having a `markup` variable holding the server-rendered markup, because it makes it clearer what's happening in the `res.render` function - e.g.:

        res.render('target', {markup, props})